sky/data/storage.py

```diff
@@ -1550,33 +1611,44 @@ def _create_gcs_bucket(self,
             f'with storage class {new_bucket.storage_class}')
         return new_bucket
 
-    def _delete_gcs_bucket(self, bucket_name: str) -> None:
+    def _delete_gcs_bucket(self, bucket_name: str) -> bool:
         """Deletes GCS bucket, including all objects in bucket
 
         Args:
           bucket_name: str; Name of bucket
+
+        Returns:
+         bool; True if bucket was deleted, False if it was deleted externally.
         """
-        try:
-            self.client.get_bucket(bucket_name)
-        except gcp.forbidden_exception() as e:
-            # Try public bucket to see if bucket exists
-            with ux_utils.print_exception_no_traceback():
-                raise PermissionError(
-                    'External Bucket detected. User not allowed to delete '
-                    'external bucket.') from e
 
-        try:
-            with log_utils.safe_rich_status(
-                    f'[bold cyan]Deleting GCS bucket {bucket_name}[/]'):
+        with log_utils.safe_rich_status(
+                f'[bold cyan]Deleting GCS bucket {bucket_name}[/]'):
+            try:
+                self.client.get_bucket(bucket_name)
```

Comment:
```suggestion
        try:
            with log_utils.safe_rich_status(
                    f'[bold cyan]Deleting GCS bucket {bucket_name}[/]'):
                    self.client.get_bucket(bucket_name)
```
Referring to `_delete_s3_bucket` and `_delete_r2_bucket`

Reply:
Ah the spinner ctx has to be outside the try block because we want one spinner to span over both, bucket get and deletion. If we put inside, then we'll need another spinner at L1642, which creates a odd break in what's shown to the user when deletion is run.